internal/commands/secrets/delete_test.go

```diff
@@ -0,0 +1,310 @@
+)
+
+func TestSecretHandler(t *testing.T) {
+	projectID := "projectID"
+	appID := "appID"
+	app := realm.App{
+		ID:          appID,
+		GroupID:     projectID,
+		ClientAppID: "eggcorn-abcde",
+		Name:        "eggcorn",
+	}
+
+	testLen := 7
+	secrets := make([]realm.Secret, testLen)
+	for i := 0; i < testLen; i++ {
+		secrets[i] = realm.Secret{
+			ID:   fmt.Sprintf("secret_id_%d", i),
+			Name: fmt.Sprintf("secret_name_%d", i),
+		}
+	}
+	for _, tc := range []struct {
+		description    string
+		testInput      []string
+		expectedOutput []secretOutput
+		expectedErr    error
+	}{
+		{
+			description: "should return successful outputs for proper secret inputs",
+			testInput:   []string{"secret_id_1"},
+			expectedOutput: []secretOutput{
+				{
+					secrets[1],
+					nil,
+				},
+			},
+		},
+		{
+			description: "should still output the errors for deletes on individual secrets",
+			testInput:   []string{"secret_id_0", "secret_id_6"},
+			expectedOutput: []secretOutput{
+				{
+					secrets[0],
+					errors.New("something happened"),
+				},
+				{
+					secrets[6],
+					errors.New("something happened"),
+				},
+			},
+			expectedErr: errors.New("something happened"),
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			realmClient := mock.RealmClient{}
+
+			var capturedAppFilter realm.AppFilter
+			realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+				capturedAppFilter = filter
+				return []realm.App{app}, nil
+			}
+
+			var capturedFindProjectID, capturedFindAppID string
+			realmClient.SecretsFn = func(groupID, appID string) ([]realm.Secret, error) {
+				capturedFindProjectID = groupID
+				capturedFindAppID = appID
+				return secrets, nil
+			}
+
+			var capturedDeleteProjectID, capturedDeleteAppID string
+
+			realmClient.DeleteSecretFn = func(groupID, appID, secretId string) error {
+				capturedDeleteProjectID = groupID
+				capturedDeleteAppID = appID
+				return tc.expectedErr
+			}
+
+			cmd := &CommandDelete{
+				inputs: deleteInputs{
+					ProjectInputs: cli.ProjectInputs{
+						Project: projectID,
+						App:     appID,
+					},
+					secrets: tc.testInput,
+				},
+				realmClient: realmClient,
+			}
+
+			assert.Nil(t, cmd.Handler(nil, nil))
+
+			for i, actual := range cmd.outputs {
+				assert.Equal(t, actual.secret, tc.expectedOutput[i].secret)
+				assert.Equal(t, actual.err, tc.expectedOutput[i].err)
+			}
+
+			assert.Equal(t, realm.AppFilter{App: appID, GroupID: projectID}, capturedAppFilter)
+			assert.Equal(t, projectID, capturedFindProjectID)
+			assert.Equal(t, appID, capturedFindAppID)
+			assert.Equal(t, projectID, capturedDeleteProjectID)
+			assert.Equal(t, appID, capturedDeleteAppID)
+		})
+	}
+
+	t.Run("should return an error", func(t *testing.T) {
+		successfulAppFn := func(filter realm.AppFilter) ([]realm.App, error) {
+			return []realm.App{app}, nil
+		}
+		successfulSecretsFn := func(groupID, appID string) ([]realm.Secret, error) {
```

Comment:
[question] are these used anywhere else? they probably can just be defined right in `clientSetup`

Reply:
they're used multiple times in the different test cases for this set